From 69dc3171284ae9c54d3932a5e8d5d8bb0d63a44c Mon Sep 17 00:00:00 2001
From: Giuseppe CAVALLARO <peppe.cavallaro@st.com>
Date: Wed, 15 Oct 2008 11:01:36 +0200
Subject: [PATCH] stmmac: review the plat_stmmacenet_data for the 7105 and 7141

Bump-up the Programmable Burst Lenght threshold.
The device can wake-up the system through the Magic frame.

Signed-off-by: Giuseppe Cavallaro <peppe.cavallaro@st.com>
Signed-off-by: Stuart Menefy <stuart.menefy@st.com>
---
 arch/sh/kernel/cpu/sh4/setup-stx7105.c |    3 ++-
 arch/sh/kernel/cpu/sh4/setup-stx7141.c |    5 +++--
 2 files changed, 5 insertions(+), 3 deletions(-)

diff --git a/arch/sh/kernel/cpu/sh4/setup-stx7105.c b/arch/sh/kernel/cpu/sh4/setup-stx7105.c
index 43932da..ff7bb3a 100644
--- a/arch/sh/kernel/cpu/sh4/setup-stx7105.c
+++ b/arch/sh/kernel/cpu/sh4/setup-stx7105.c
@@ -497,7 +497,7 @@ static void fix_mac_speed(void* priv, unsigned int speed)
 
 static struct plat_stmmacenet_data stx7105eth_private_data = {
 	.bus_id = 0,
-	.pbl = 8,
+	.pbl = 32,
 	.has_gmac = 1,
 	.fix_mac_speed = fix_mac_speed,
 };
@@ -520,6 +520,7 @@ static struct platform_device stx7105eth_device = {
         	},
 	},
 	.dev = {
+		.power.can_wakeup = 1,
 		.platform_data = &stx7105eth_private_data,
 	}
 };
diff --git a/arch/sh/kernel/cpu/sh4/setup-stx7141.c b/arch/sh/kernel/cpu/sh4/setup-stx7141.c
index 138b4b3..efab94d 100644
--- a/arch/sh/kernel/cpu/sh4/setup-stx7141.c
+++ b/arch/sh/kernel/cpu/sh4/setup-stx7141.c
@@ -570,13 +570,13 @@ static void fix_mac_speed(void *priv, unsigned int speed)
 static struct plat_stmmacenet_data stx7141eth_private_data[2] = {
 {
 	.bus_id = 0,
-	.pbl = 8,
+	.pbl = 32,
 	.has_gmac = 1,
 	.fix_mac_speed = fix_mac_speed,
 	.bsp_priv = 0,
 }, {
 	.bus_id = 1,
-	.pbl = 8,
+	.pbl = 32,
 	.has_gmac = 1,
 	.fix_mac_speed = fix_mac_speed,
 	.bsp_priv = 1,
@@ -621,6 +621,7 @@ static struct platform_device stx7141eth_devices[2] = {
 		},
 	},
 	.dev = {
+		.power.can_wakeup = 1,
 		.platform_data = &stx7141eth_private_data[1],
 	}
 } };
-- 
1.5.3.6

